learn/using-contracts/setup.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Setup your Node Client
+order: 1
+---
+
+# Introduction
+
+This is a simple tutorial showing you how to use of powerful node REPL to interact with
+a CW20 token contract (fungible tokens, like ERC20) on [coralnet](https://github.com/CosmWasm/testnets/tree/master/coralnet).
+
+I will walk you through uploading contract code and creating a concrete instance (the same `cw20-base`
+wasm code can be reused to create dozens of token contracts with different symbols and distributions).
+Then I will show you how to easily interact with this contract. As JSON manipulation and local variables
+are not so much fun in BASH, we use the [`@cosmjs/cli`](https://github.com/CosmWasm/cosmjs/tree/master/packages/cli)
+tool instead of the `coral` CLI tool.
+
+But, before we get into the fun part of playing with the smart contracts, I want to make sure
+you know how to use your tools and not loose your private keys.
+
+You must have Node 10+ installed locally to run it. It has been tested on Ubuntu, and may not
+work properly on Windows (we assume a HOME environmental variable). PRs welcome.
+
+## Connecting to the chain
+
+The first step before doing anything is ensuring we can create an account and connect to the chain.
+You will always use the following command to start up the `@cosmjs/cli` with some cw20-specific helpers preloaded
+(in addition to all the general helpers it has).
+
+```shell
+npx @cosmjs/cli --init https://raw.githubusercontent.com/CosmWasm/cosmwasm-plus/master/contracts/cw20-base/helpers.ts
```

Comment:
I really like this `helpers` existence. Very useful. Maybe we can add embed contract helpers URL inside tx msg like source and builder. And in the wasm.glass as well 
![image](https://user-images.githubusercontent.com/11277600/90223271-6c8c4400-de16-11ea-8ee8-4541710afef6.png)

Reply:
Nice idea. We should make this more extensible somehow.

There was already the request to upload all json schemas (and maybe all source) along with the wasm.
A URL for some cosmjs helpers is a good idea.

What do you think @webmaster128 ?

Reply:
Allowing untrusted "helpers" besically means contract uploaders can trick a user to execute any command they want on their local machine (such as uploading `~/.ssh/id_rsa` to a server or do nasty things with your metamask storage). 

What I could imagine having is an autogenerated helpers script that is generated by a trusted service like wasm.glass, which is a template with only few well validated variables.

Reply:
I just made an issue to discuss this more: https://github.com/CosmWasm/wasmd/issues/251

Feel free to comment there